Gordon360/Controllers/RecIM/SportsController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Services.RecIM;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Gordon360.Controllers.RecIM
+{
+    [Route("api/recim/[controller]")]
+    [AllowAnonymous]
+    public class SportsController : GordonControllerBase
+    {
+        private readonly ISportService _sportService;
+
+        public SportsController(ISportService sportService)
+        {
+            _sportService = sportService;
+        }
+
+        [HttpGet]
+        [Route("")]
+        public ActionResult<IEnumerable<SportViewModel>> GetSports()
+        {
+            var res = _sportService.GetSports();
+            return Ok(res);
+        }
+
+        [HttpGet]
+        [Route("{sportID}")]
+        public ActionResult<SportViewModel> GetSportByID(int sportID)
+        {
+            var res = _sportService.GetSportByID(sportID);
+            return Ok(res);
+        }
+      
+        [HttpPatch]
+        [Route("{sportID}")]
+        public async Task<ActionResult> UpdateSport(int sportID, SportViewModel updatedSport)
```

Comment:
Should this be a `SportUploadViewModel`, same as for the POST route? I imagine that the `SportUploadViewModel` does not have a sportID and that would mean updating the service method as well to take in a sportID (see below)

Reply:
Additionally, perhaps a HTTP PATCH route method should be called `PatchSport(...)`?

Reply:
> Additionally, perhaps a HTTP PATCH route method should be called `PatchSport(...)`?

I don't think we need to tie the method names to the HTTP method used. For example, a `POST` route could be `Update Involvement` or `Upload Involvement Image`.

`UpateSport` describes this action well, and I don't think `PatchSport` is any clearer.

Reply:
Well technically they are both different forms of "updating" a resource, yes. But it would be clearer to have something distinct to notate that this is a different kind of operation from our other Update...() methods. This is an update of particular fields, not a full replacement.

Our real issue is that all of our previous "Update...()" methods should really be called "Replace...()" because they are replacing resources in a particular location, but that is a problem not worth fixing at this point.

I guess the problem is not yours to fix @amos-cha, so feel free to disregard then.

Reply:
I mean if there is a API wide cleanup, I'll make that change before the merge into develop down the road. Just keep me posted on it :)

Reply:
No, I think it would just be more confusing if you did something else now. Technically, your methods are named with proper terminology, it would be the routes outside of RecIM that would need updating.